oci/spec_opts_windows_test.go

```diff
@@ -108,3 +113,186 @@ func TestWithWindowNetworksAllowUnqualifiedDNSQuery(t *testing.T) {
 		}
 	}
 }
+
+func newFakeExtendedImage(config imageExtended) (Image, error) {
```

Comment:
I was looking at the `imageExtended` and `imageConfigExtended` types; looks like `imageExtended` is only used to detect the `ArgsEscaped` bool (for which it doesn't need any of the other fields),
but to generate a JSON blob in the tests, `imageConfigExtended` is needed. Unfortunately, this currently means copying the `Entrypoint` and `Cmd` to make that work.

A possible alternative could be to embed the `v1.ImageConfig` (at least that doesn't duplicate those fields then, and less risk of things diverging?);

```go
// imageExtended is a subset of the oci Image struct that changes the Config
// type to include `ArgsEscaped` which is not an OCI field, but is supported by
// Docker built images. This allows for proper support of the 'shell form'
// of Entrypoint or Cmd from a Dockerfile for Windows containers. Please see
// WithImageConfigArgs windows assignment for more information.
type imageExtended struct {
	Config struct {
		v1.ImageConfig
		ArgsEscaped bool `json:"ArgsEscaped,omitempty"`
	}
}
```

The `newFakeExtendedImage()` utility could then accept a regular `ocispec.ImageConfig`, and set the `ArgsEscaped` internally;

```go
func newFakeExtendedImage(config ocispec.ImageConfig) (Image, error) {
	im := imageExtended{}
	im.Config.ImageConfig = config
	im.Config.ArgsEscaped = true

	configBlob, err := json.Marshal(im)
	if err != nil {
		return nil, err
	}
	configDescriptor := ocispec.Descriptor{
		MediaType: ocispec.MediaTypeImageConfig,
		Digest:    digest.NewDigestFromBytes(digest.SHA256, configBlob),
	}

	return fakeImage{
		config: configDescriptor,
		blobs: map[string]blob{
			configDescriptor.Digest.String(): configBlob,
		},
	}, nil
}
```

This (very slightly) simplified the tests as well, e.g. it would become something like;

```go
img, err := newFakeExtendedImage(ocispec.ImageConfig{
    Entrypoint: []string{`powershell.exe -Command "C:\My Data\MyExe.exe" -arg1 "-arg2 value2"`},
    Cmd:        []string{`cmd.exe /S /C "C:\test path\test.exe"`},
})
```

Reply:
(disclaimer: I gave it a quick try on my machine, but wasn't fully able to test it, because I'm not on a Windows machine)

Reply:
I did it this way to avoid deserializing all the extra fields we don't actually need again on the hot path. But I take the point, I can do this as two separate structs actually, one for the WithImage code and one for setting the blob in the tests. Will fix

Reply:
> I did it this way to avoid deserializing all the extra fields we don't actually need again on the hot path.

Ah, yes, fair point

>  I can do this as two separate structs actually, one for the WithImage code and one for setting the blob in the tests.

Thanks!